core/app/models/spree/order/checkout.rb

```diff
@@ -221,7 +221,15 @@ def update_from_params(params, permitted_params, request_env = {})
             run_callbacks :updating_from_params do
               # Set existing card after setting permitted parameters because
               # rails would slice parameters containg ruby objects, apparently
-              existing_card_id = @updating_params[:order] ? @updating_params[:order].delete(:existing_card) : nil
+
+              payment_method = if @updating_params.dig(:order, :payments_attributes)
+                                 id = @updating_params[:order][:payments_attributes].first[:payment_method_id]
+                                 Spree::PaymentMethod.find(id)
+                               end
+
+              existing_card_id = if payment_method&.type&.include?('Spree::Gateway')
```

Comment:
@agnieszkajacek this check cannot be declared this way - it will make it impossible to work with 3rd party extension eg. Spree braintree vzero